app/models/Restaurant.java

```diff
@@ -11,45 +13,81 @@
     private String name;
 
-    @OneToOne
+    @OneToOne(cascade = CascadeType.ALL)
     @PrimaryKeyJoinColumn(referencedColumnName = "restaurantId")
-    @JsonView(RestaurantViews.BasicDetails.class)
+    @JsonView(View.BasicDetails.class)
     private Address address;
 
-    @OneToOne
+    @OneToOne(cascade = CascadeType.ALL)
     @PrimaryKeyJoinColumn(referencedColumnName = "restaurantId")
-    @JsonView(RestaurantViews.AllDetails.class)
+    @JsonView(View.AllDetails.class)
     private Coordinates coordinates;
 
     @Column(columnDefinition = "BIGINT")
-    @JsonView(RestaurantViews.BasicDetails.class)
+    @JsonView(View.BasicDetails.class)
     private long phone;
     @Column(name = "workingHours",length = 20)
-    @JsonView(RestaurantViews.BasicDetails.class)
+    @JsonView(View.BasicDetails.class)
     private String workingHours;
-    @JsonView(RestaurantViews.BasicDetails.class)
+    @JsonView(View.BasicDetails.class)
     private double rating;
     @Column(name = "reservationPrice")
-    @JsonView(RestaurantViews.AllDetails.class)
+    @JsonView(View.AllDetails.class)
     private double reservationPrice;
     @Column(length = 200)
-    @JsonView(RestaurantViews.AllDetails.class)
+    @JsonView(View.AllDetails.class)
     private String deals;
-    @JsonView(RestaurantViews.BasicDetails.class)
+    @JsonView(View.BasicDetails.class)
     private String image;
-
     
     public static Model.Finder<String, Restaurant> find = new Model.Finder<String, Restaurant>(String.class, Restaurant.class);
 
+    /**
+     * Validates the restaurant properties.
+     *
+     * @return true if validation is successful
+     */
+    @Override
+    public boolean isValid() {
+        return !name.equals("") && !address.getCity().equals("") && !address.getCountry().equals("") &&
+                !address.getStreetName().equals("") && coordinates.getLatitude() != 0 && coordinates.getLongitude() != 0
+                && phone != 0;
+    }
+
+    public void updateCoordinates(Coordinates coordinates){
+        this.getCoordinates().setLatitude(coordinates.getLatitude());
+        this.getCoordinates().setLongitude(coordinates.getLongitude());
+    }
+
+    public void updateAddress(Address address) {
+        this.getAddress().setStreetName(address.getStreetName());
+        this.getAddress().setCity(address.getCity());
+        this.getAddress().setCountry(address.getCountry());
+    }
+
+    public void updateData(Restaurant restaurant) {
+        this.setDeals(restaurant.getDeals());
+        this.setImage(restaurant.getImage());
+        this.setName(restaurant.getName());
+        this.setPhone(restaurant.getPhone());
+        this.setRating(restaurant.getRating());
+        this.setReservationPrice(restaurant.getReservationPrice());
+        this.setWorkingHours(restaurant.getWorkingHours());
+
+        this.updateAddress(restaurant.getAddress());
+        this.updateCoordinates(restaurant.getCoordinates());
+        this.save();
```

Comment:
This is wrong. You should never perform db calls from the model. Model is used only to prepare object for update! Please, use persistence manager for save operation.